tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java

```diff
@@ -135,17 +138,18 @@ public Integer call() throws Exception {
         // Sorted map of block height to StoredBlock object.
         final TreeMap<Integer, StoredBlock> checkpoints = new TreeMap<>();
 
-        long now = new Date().getTime() / 1000;
-        peerGroup.setFastCatchupTimeSecs(now);
+        Instant now = TimeUtils.currentTime();
+        peerGroup.setFastCatchupTime(now);
 
-        final long timeAgo = now - (86400 * days);
-        System.out.println("Checkpointing up to " + TimeUtils.dateTimeFormat(timeAgo * 1000));
+        Instant timeAgo = now.minus(days, ChronoUnit.DAYS);
+        System.out.println("Checkpointing up to " + TimeUtils.dateTimeFormat(timeAgo.toEpochMilli()));
 
         chain.addNewBestBlockListener(Threading.SAME_THREAD, block -> {
             int height = block.getHeight();
-            if (height % params.getInterval() == 0 && block.getHeader().getTimeSeconds() <= timeAgo) {
+            if (height % params.getInterval() == 0 && block.getHeader().getTimeInstant().compareTo(timeAgo) <= 0) {
```

Comment:
I would use `isBefore()` or `isAfter()` here. Although using `compareTo` and `<=` looks like it's closer to the old code/easier to compare, it's actually different (I briefly thought the `0` value was replacing `timeAgo` with epoch seconds.)

Reply:
Ok, I'll rewrite it with `isAfter()` (and the operands swapped).